### ISSUE TITLE
java_binary: superfluous dependency on MANIFEST from runfiles

### ISSUE THREAD

**Issue by scaiper** [bazelbuild/bazel#21708]
### Description of the bug:

Target `name`_deploy.jar of the java_binary rule implicitly depends on MANIFEST file containing absolute paths to various files. That in turn makes build non remote cacheable, as this file is different on every host.

It looks like this manifest files is not actually needed for action in question, as it only appears in Inputs, but not in Command line.

### Which category does this issue belong to?

Java Rules

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

1. Create a build target using java_binary.
2. Run `bazel aquery <your rule>_deployjars_internal_rule`

Notice that MANIFEST appears in `Inputs` but not in `Command Line`.

### Which operating system are you running Bazel on?

linux

### What is the output of `bazel info release`?

release 7.0.2

### Have you found anything relevant by searching the web?

No, I have not.

### Any other information, logs, or outputs that you want to share?

Manifest contains lines like this.
```
_main/external/java_contribs_stable/com/fasterxml/jackson/core/jackson-annotations/2.16.1/processed_jackson-annotations-2.16.1.jar /home/scaiper/.cache/bazel/_bazel_scaiper/9f4abd0e97231642b610412fa4299bc8/execroot/_main/bazel-out/k8-fastbuild/bin/external/java_contribs_stable/com/fasterxml/jackson/core/jackson-annotations/2.16.1/processed_jackson-annotations-2.16.1.jar
```

**Comment by pcjanzen** [comment]
Can you create a reproducer using `--disk_cache`?

The manifest file is specifically excluded when creating the hash of the action that's looked up in a remote cache.

**Comment by scaiper** [comment]
We have a remote cache configured.
But I've found this commits in master https://github.com/bazelbuild/bazel/commit/6a4d61ebf2dc2214ffe6abc807e4f597175e1a6c https://github.com/bazelbuild/bazel/commit/cff79d0aa6f2d470b8648f2b89beeefd0867e177 . Looks like it fixes the issue.

**Comment by fmeum** [comment]
@bazel-io fork 7.1.1

**Comment by fmeum** [comment]
@lberki Do you think we need to cherry-pick the fixes mentioned in https://github.com/bazelbuild/bazel/issues/21708#issuecomment-2000060938?

**Comment by mattnworb** [comment]
Thanks for creating this issue and finding the linked commits, we happened to just run into this issue today in our CI environment which happens to clone the repo-being-built to a path containing a random identifier for each new build, causing all these actions to be cache misses. 

Just wanted to share that we were able to work around this by setting `--output_base` to a static value, although that might not work so well for solving this across different developer machines.

**Comment by lberki** [comment]
@scaiper good sleuthing! I was wondering why those are not enough but then I noticed the version number.

@fmeum it depends on the impact; those two changes broke some minor things at Google, so it's not a "free cherry-pick", but if it's painful enough for someone, I'd be fine with cherry-picking them to 7.2 .

**Comment by fmeum** [comment]
@iancha1992 Could you cherry-pick the two commits mentioned in https://github.com/bazelbuild/bazel/issues/21708#issuecomment-2000060938 into 7.2.0?

**Comment by fmeum** [comment]
@bazel-io fork 7.2.0

**Comment by hvadehra** [comment]
> so it's not a "free" cherry-pick

@lberki Alos, I thought these were only possible because of your earlier series of `RunfilesSupport` changes (that I think did not make it into 7.0.2)? Or are they safe independently?

**Comment by lberki** [comment]
They are safe independency as far as I can tell, modulo merging trouble they just remove functionality (an artifact from the inputs of an action), not add anything.

@hvadehra are you fine with cherrypicking these into 7.2.0 ?

**Comment by hvadehra** [comment]
Yes, SGTM

**Comment by lberki** [comment]
@iancha1992 then please do cherry-pick it! I'd love to do away with this particular quirk.

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.2.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.2.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=7.2.0rc1. Thanks!
